types/modbus/data.pp

```diff
@@ -0,0 +1,2 @@
+#
+type Collectd::Modbus::Data = Struct[{Optional['instance'] => String, NotUndef['type'] => String[1], NotUndef['register_base'] => Numeric, NotUndef['register_type'] => String[1]}]
```

Comment:
Could we add some tests for this and the other types.

e.g. like https://github.com/voxpupuli/puppet-systemd/tree/master/spec/type_aliases